config.go

```diff
@@ -1,10 +1,60 @@
 package aibridge
 
+import "go.uber.org/atomic"
+
 type ProviderConfig struct {
```

Comment:
This is a lot especially given that the baseURL and key won't change during a provider's lifetime within Coder.

Have you considered moving the methods for changing upstream logging settings to the provider rather than having the config be mutable? In Coder for example, we don't change the `coderd.Options` type after passing it to the API.